File: src/main/java/no/nav/sosialhjelp/soknad/web/selftest/domain/Selftest.java
//package no.nav.sosialhjelp.soknad.web.selftest.domain;
//
//import java.util.Collections;
//import java.util.List;
//
//import static java.util.Optional.ofNullable;
//
///*
//Kopiert inn fra no.nav.sbl.dialogarena:common-web
//Endringer gjort i no.nav.common:web gjør at vi heller benytter den fra det gamle artefaktet.
//Kan mest sannsynlig oppgraderes, hvis vi får selftest til å fungere fra no.nav.common:web
//*/
//
//public class Selftest {
//    private String application;
//    private String version;
//    private String timestamp;
//    private int aggregateResult;
//    private List<SelftestEndpoint> checks;
//
//    public String getApplication() {
//        return application;
//    }
//
//    public Selftest setApplication(String application) {
//        this.application = application;
//        return this;
//    }
//
//    public String getVersion() {
//        return version;
//    }
//
//    public Selftest setVersion(String version) {
//        this.version = version;
//        return this;
//    }
//
//    public String getTimestamp() {
//        return timestamp;
//    }
//
//    public Selftest setTimestamp(String timestamp) {
//        this.timestamp = timestamp;
//        return this;
//    }
//
//    public int getAggregateResult() {
//        return aggregateResult;
//    }
//
//    public Selftest setAggregateResult(int aggregateResult) {
//        this.aggregateResult = aggregateResult;
//        return this;
//    }
//
//    public List<SelftestEndpoint> getChecks() {
//        return ofNullable(checks).orElseGet(Collections::emptyList);
//    }
//
//    public Selftest setChecks(List<SelftestEndpoint> checks) {
//        this.checks = checks;
//        return this;
//    }
//}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/health/selftest/Selftest.kt
package no.nav.sosialhjelp.soknad.health.selftest

/*
Kopiert inn fra no.nav.sbl.dialogarena:common-web
Endringer gjort i no.nav.common:web gjør at vi heller benytter den fra det gamle artefaktet.
Kan mest sannsynlig oppgraderes, hvis vi får selftest til å fungere fra no.nav.common:web
*/
data class Selftest(
    val application: String? = null,
    val version: String? = null,
    val timestamp: String? = null,
    val aggregateResult: Int = 0,
    private val checks: List<SelftestEndpoint> = emptyList()
) {
    fun getChecks(): List<SelftestEndpoint> {
        return checks
    }
}

data class SelftestEndpoint(
    val endpoint: String?,
    val description: String?,
    val errorMessage: String?,
    private val result: Int?,
    val responseTime: String?,
    val stacktrace: String?,
    val critical: Boolean = false,
) {
    fun harFeil(): Boolean {
        return this.result != SelftestService.STATUS_OK
    }

    fun getResult(): Int {
        return result ?: SelftestService.STATUS_ERROR
    }
}
